payara/monitoring-console
# FISH-481 QoL Improvments

### Summary
PR for various QoL improvements for Payara InSight.

- [x] adds a warning icon to "danger zone" dialogues
- [x] smoother "Add page" flow 
- [x] renaming a Watch should not leave Watch with old name behind
- [x] font size adjustments
- [x] auto fill empty cells (+ option)

### Testing
Manual verification that changes do what they should and don't break something.

#### Testing Performed
For Warning Icon:
* select a widget
* click _Remove Widget_ button in the settings
* check the dialogue has a warning icon
* open _Application_ settings, tab _Pages_
* click export pages to create a file with some pages in it
* click import pages and select the exported file
* check the import dialogue has a warning icon

For smoother add page flow:
* create a new page by clicking _Add New Page_ button
* enter a page name in the popup dialogue (notice the input is focused so you can start typing directly after pressing the button)
* confirm the entered page name with _Enter_ key and check this closes the dialogue and creates the page
* check the blank page has a plus button shown that basically spans the full page which when pressed opens the _Add Widget_ dialogue
* change to more columns using the buttons in the left sidebar
* check each column shows the plus button
* use the plus button to actually add a widget and check it has the intended effect

For renaming a watch should not leave behind a watch with the old name:
* open watch settings as user or administrator
* copy a system watch
* check only creation of a new watch is confirmed with a feedback message
* edit the created watch and change the name
* check saving the watch with the new name is confirmed and removing the watch with the old name is confirmed as well

For font sizes:
* use developer tools to check the used font sizes are 16px for normal text and secondary headings, 24px for primary headings (note there are some exceptions, see FISH-521 comment)

For auto fill empty cells (1 column only):
* open page settings
* check new options _Fill Empty Cells_ exists
* switch options to _Yes_ (for example for _Core_ page)
* switch page to 2 column layout and check the 2nd widget in 1st column now spans 3 rows
* switch page to 4 column layout, move one of the widgets in 3rd column to 4th and check both 3rd and 4th column widgets now span 2 rows
* switch back to 3 column layout and check all widgets don't span any rows and columns
* switch to 1 column layout and check all widgets span 1 row (same height)

### Documentation
https://github.com/payara/Payara-Enterprise-Documentation/pull/76

### Release
When merged new release of InSight should be done. As usual changes only affect the implementation side. Version increments similar to last release are needed.
